### PR TITLE
adding default values to near/far clipping planes

### PR DESCRIPTION
**Pull request by winthos**
In regards to 3rd party cameras, this fixes an issue where light and shadow resolutions were dynamically scaled based on the clipping plane values, so defaulting this to the same clipping planes as the agent camera for parity

### CODE REVIEW

File: unity/Assets/Scripts/AgentManager.cs [hunk 1]
```diff
@@ -471,7 +488,9 @@ private enum serverTypes { WSGI, FIFO };
         float fieldOfView = DEFAULT_FOV,
         string skyboxColor = null,
         bool orthographic = false,
-        float? orthographicSize = null
+        float? orthographicSize = null,
+        float? nearClippingPlane = null,
+        float? farClippingPlane = null
```
Comment (mattdeitke) [inline unity/Assets/Scripts/AgentManager.cs:493]: It looks like these do not need to be optional `float?`, since they are being parsed as:
```c#
        if (farClippingPlane == null) {
            camera.farClipPlane = 20.0f;
        } else {
            camera.farClipPlane = (float) farClippingPlane;
        }
```
can:

- `float? farClippingPlane = null` become `float farClippingPlane = 20.0f`
- `float? nearClippingPlane = null` become `float nearClippingPlane = 0.1f`

File: unity/Assets/Scripts/AgentManager.cs [hunk 2]
```diff
@@ -419,7 +419,9 @@ private enum serverTypes { WSGI, FIFO };
         float fieldOfView,
         string skyboxColor,
         bool? orthographic,
-        float? orthographicSize
+        float? orthographicSize,
+        float? nearClippingPlane = null,
+        float? farClippingPlane = null
```
Comment (mattdeitke) [inline unity/Assets/Scripts/AgentManager.cs:424]: Can
```c#
        float? nearClippingPlane = null,
        float? farClippingPlane = null
```
not have defaults for the private method?

Perhaps this is just a personal style preference, but since this is a private method, I generally like how every single call to it must specify all parameters.

Here, if new parameters are added, like `nearClippingPlane` and `farClippingPlane`, they have to explicitly be passed in at every place the private method is called, and if they're not, a compiler warning will appear.

In short, I think this helps ensure that public parameters are always passed to the private method.

**Review by mattdeitke — CHANGES_REQUESTED** [review]

File: unity/Assets/Scripts/AgentManager.cs (hunk 1 above)
Comment (winthos) [reply unity/Assets/Scripts/AgentManager.cs]: ok check the new commit, made changes

File: unity/Assets/Scripts/AgentManager.cs (hunk 2 above)
Comment (winthos) [reply unity/Assets/Scripts/AgentManager.cs]: sounds good, changes in

**Review by mattdeitke — APPROVED** [review]
Looks good. The only other thing I'd say is to autoformat it, but I'm not sure if there's going to be a more automatic process for that.

**Comment by ekolve** [comment]
It should be possible to format the block using VSCode if you highlight and select Format Selection.  VSCode will respect the .editorconfig that is checked in (or at least it should).

File: unity/Assets/Scripts/AgentManager.cs [hunk 3]
```diff
@@ -442,6 +444,21 @@ private enum serverTypes { WSGI, FIFO };
...
+            camera.nearClipPlane = 0.1f;
+        }
+
+        else { camera.nearClipPlane = (float) nearClippingPlane;}
+
+        if(farClippingPlane == null) {
+            camera.farClipPlane = 20.0f;
+        }
+
+        else { camera.farClipPlane = (float) farClippingPlane;}
```
Comment (mattdeitke) [inline unity/Assets/Scripts/AgentManager.cs:460]: Wait, one other bug (appears twice, also for `nearClipPlane`)... If these are `null` (from UpdateThirdPartyCameras), then they should be keeping their current values.

Can
```c#
        if (farClippingPlane == null) {
            camera.farClipPlane = 20.0f;
        } else {
            camera.farClipPlane = (float) farClippingPlane;
        }
```

then be changed to
```c#
        if (farClippingPlane != null) {
            camera.farClipPlane = (float) farClippingPlane;
        }
```

Comment (winthos) [reply unity/Assets/Scripts/AgentManager.cs]: gotchya, changed with latest commit. Also formatted all changes (Format Selection works flawlessly)

File: unity/Assets/Scripts/AgentManager.cs [hunk 4]
```diff
@@ -471,7 +484,9 @@ private enum serverTypes { WSGI, FIFO };
         float fieldOfView = DEFAULT_FOV,
         string skyboxColor = null,
         bool orthographic = false,
-        float? orthographicSize = null
+        float? orthographicSize = null,
+        float nearClippingPlane = 0.1f,
+        float farClippingPlane = 20.0f
```
Comment (ekolve) [inline unity/Assets/Scripts/AgentManager.cs:489]: Instead of hardcoding these values can we instead grab them from this.primaryAgent.m_Camera.nearClipPlan/farClipPlane within the method if the passed in value is null?

**Review by ekolve — CHANGES_REQUESTED** [review]

File: unity/Assets/Scripts/AgentManager.cs (hunk 4 above)
Comment (winthos) [reply unity/Assets/Scripts/AgentManager.cs]: updated with the latest commit

**Review by ekolve — APPROVED** [review]
